File: register_test.py
import json
import urllib.request
import urllib.error

url = 'http://127.0.0.1:8000/api/register'
payload = {
    'first_name': 'Test',
    'last_name': 'User',
    'middle_name': 'Q',
    'email': 'testuser12345@example.com',
    'password': 'password123',
    'password_confirmation': 'password123',
}
data = json.dumps(payload).encode('utf-8')
request = urllib.request.Request(url, data=data, headers={'Content-Type': 'application/json'})

try:
    with urllib.request.urlopen(request) as response:
        print('STATUS', response.status)
        print(response.read().decode('utf-8'))
except urllib.error.HTTPError as error:
    print('STATUS', error.code)
    print(error.read().decode('utf-8'))

File: register-test.js
const http = require('http');
const payload = JSON.stringify({
  first_name: 'Test',
  last_name: 'User',
  middle_name: 'Q',
  email: `testuser${Date.now()}@example.com`,
  password: 'password123',
  password_confirmation: 'password123',
});

const options = {
  hostname: '127.0.0.1',
  port: 8000,
  path: '/api/register',
  method: 'POST',
  headers: {
    'Content-Type': 'application/json',
    'Content-Length': Buffer.byteLength(payload),
  },
};

const req = http.request(options, (res) => {
  let data = '';
  res.on('data', (chunk) => (data += chunk));
  res.on('end', () => {
    console.log('statusCode', res.statusCode);
    console.log(data);
  });
});

req.on('error', (error) => {
  console.error(error);
});

req.write(payload);
req.end();
